lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1763,6 +1812,75 @@ selectSeqStatePendingIxs wid =
   where
     fromRes = fmap (W.Index . seqStatePendingIxIndex . entityVal)
 
+insertScriptPool
+    :: forall k . WalletKey k
+    => W.WalletId
+    -> W.SlotNo
+    -> Seq.VerificationKeyPool k
+    -> SqlPersistT IO ()
+insertScriptPool wid sl pool = do
+    void $ dbChunked insertMany_
+        [ SeqStateAddress wid sl (W.Address payload) (getIndex ix) W.MultisigScript state
+        | (KeyHash payload, (ix, state) )
+        <- Map.toList (Seq.verPoolIndexedKeys pool)
+        ]
+    void $ dbChunked insertMany_ $
+        concatMap toDB $ Map.toList (Seq.verPoolKnownScripts pool)
+  where
+    toAddress = W.Address . xpubToBytes . getRawKey
+    toDB (scriptHash, verKeys) =
+        zipWith (SeqStateScriptHash wid sl scriptHash . toAddress) verKeys [0..]
+
+instance Ord ScriptHash where
+    compare (ScriptHash sh1) (ScriptHash sh2) = compare sh1 sh2
+
+selectScriptPool
+    :: forall k .
+        ( SoftDerivation k
+        , WalletKey k
+        )
+    => W.WalletId
+    -> W.SlotNo
+    -> Seq.AddressPoolGap
+    -> k 'AccountK XPub
+    -> SqlPersistT IO (Seq.VerificationKeyPool k)
+selectScriptPool wid sl gap xpub = do
+    verKeys <- fmap entityVal <$> selectList
+        [ SeqStateAddressWalletId ==. wid
+        , SeqStateAddressSlot ==. sl
+        , SeqStateAddressRole ==. W.MultisigScript
+        ] [Asc SeqStateAddressIndex]
+    scripts <- fmap entityVal <$> selectList
+        [ SeqStateScriptHashWalletId ==. wid
+        , SeqStateScriptHashSlot ==. sl
+        ] []
+    pure $ scriptPoolFromEntities verKeys scripts
+  where
+    verKeyMap =
+        Map.fromList .
+        map (\x -> ( coerce $ seqStateAddressAddress x
+                   , ( W.Index $ seqStateAddressIndex x
+                     , seqStateAddressStatus x
+                     )))
+    insertIfNotNothing k xpubM i = case xpubM of
+        Just xpub' -> Map.insertWith (++) k [(i, liftRawKey xpub')]
+        Nothing -> id
+    knownScripts =
+        Map.map (map snd . sortOn fst) .
+        foldr
+        ((\ (sh, vK, i) -> insertIfNotNothing sh vK i)
+         . (\ x -> (  seqStateScriptHashScriptHash x
+                    , xpubFromBytes $ W.unAddress $
+                      seqStateScriptHashVerificationKey x
+                    , seqStateScriptHashIndex x)))
+        Map.empty
+    scriptPoolFromEntities
+        :: [SeqStateAddress]
+        -> [SeqStateScriptHash]
+        -> Seq.VerificationKeyPool k
+    scriptPoolFromEntities verKeys scripts
+        = Seq.mkVerificationKeyPool xpub gap (verKeyMap verKeys) (knownScripts scripts)
```

Comment:
:point_up: note to self, I haven't checked that. To do after revising the database table schema.